scipy/optimize/optimize.py

```diff
@@ -2048,16 +2051,16 @@ def golden(func, args=(), brack=None, tol=_epsilon, full_output=0):
     interval.
 
     """
-    options = {'xtol': tol}
+    options = {'xtol': tol, 'maxiter': maxiter}
     res = _minimize_scalar_golden(func, brack, args, **options)
     if full_output:
-        return res['x'], res['fun'], res['nfev']
+        return res['x'], res['fun'], res['nfev'], res['nit']
```

Comment:
This is still bw incompatible. It's not a matter of element order, the size of returned tuple should also not be changed. Note that this is not a big deal since calling `golden` is actually loosely deprecated over calling `minimize_scalar(..., method='golden')` and the latter returns an `OptimizeResult` which can be changed in a bw compatible way (as you already did).